src/uu/fmt/src/fmt.rs

```diff
@@ -141,21 +137,40 @@ fn parse_arguments(args: impl uucore::Args) -> UResult<(Vec<String>, FmtOptions)
                 ));
             }
         };
-        if fmt_opts.width > MAX_WIDTH {
-            return Err(USimpleError::new(
-                1,
-                format!(
-                    "invalid width: '{}': Numerical result out of range",
-                    fmt_opts.width,
-                ),
-            ));
-        }
-        fmt_opts.goal = cmp::min(
-            fmt_opts.width * DEFAULT_GOAL_TO_WIDTH_RATIO / 100,
-            fmt_opts.width - 3,
-        );
     };
 
+    match matches.get_one::<String>(OPT_GOAL) {
+        Some(s) => {
+            fmt_opts.goal = match s.parse::<usize>() {
+                Ok(t) => t,
+                Err(e) => {
+                    return Err(USimpleError::new(
+                        1,
+                        format!("Invalid GOAL specification: {}: {}", s.quote(), e),
+                    ));
+                }
+            };
+            if !matches.contains_id(OPT_WIDTH) {
+                // GNU fmt.c:407
+                fmt_opts.width = fmt_opts.goal + 10;
+            }
+        }
+        // GNU fmt.c:411 equals to 0.935 = 93.5%
+        None => fmt_opts.goal = fmt_opts.width * (2 * (100 - 7) + 1) / 200,
+    }
+    if fmt_opts.goal > fmt_opts.width {
+        return Err(USimpleError::new(1, "GOAL cannot be greater than WIDTH."));
+    }
```

Comment:
These lines seem to do the same as lines 174 - 192, just using a `match` instead of an `if let`. As your code with the `match` looks cleaner, I would keep it and remove the `if let` code instead.